features/events/syslog/src/main/java/org/opennms/netmgt/syslogd/ConvertToEvent.java

```diff
@@ -148,7 +154,20 @@ public static final EventBuilder toEventBuilder(SyslogMessage message, String sy
         // Add any syslog message parameters as event parameters.
         message.getParameters().forEach((k, v) -> bldr.addParam(k.toString(), v));
 
-        final InetAddress hostAddress = message.getHostAddress();
+        InetAddress hostAddress = message.getHostAddress();
+        String hostIpAddress = null;
+        if (hostAddress == null &&
+                message.getHostName() != null &&
+                !LocationUtils.isDefaultLocationName(location) &&
+                locationAwareDnsLookupClient != null) {
+            CompletableFuture<String> future = locationAwareDnsLookupClient.lookup(message.getHostName(), location, systemId);
+            try {
+                hostIpAddress = future.get();
```

Comment:
This is too big of a performance hit for this code as it stands.

The requirement for this has typically been that the host given in the syslog message must match the node label for them to be associated.

Reply:
Use the same cache that is used for flow reverse lookups. That code can handle this with ease and is very performant

Reply:
Added localCache for syslog.

Reply:
> Use the same cache that is used for flow reverse lookups. That code can handle this with ease and is very performant

That's not designed for querying across locations at least for now.